src/time-picker/time-picker.jsx

```diff
@@ -159,15 +198,19 @@ const TimePicker = React.createClass({
   },
 
   openDialog() {
-    this.setState({
-      dialogTime: this.getTime(),
-    });
+    if (this.state.time) {
```

Comment:
or we could fix the `componentWillReceiveProps` of the `Clock` component to use:
```jsx
    this.setState({
      muiTheme: newMuiTheme,
      selectedTime: nextProps.initialTime  || new Date(),
    });
```